Statistics/milestone 3

urweakness opened the pull request:
## README
Стек: SwiftUI + MVVM
Декомпозиция на доске: [README](https://github.com/users/pavelseleznev/projects/1/views/11)
Декомпозиция эпика: [table teamwork (view)](https://github.com/users/pavelseleznev/projects/1/views/7?filterQuery=label%3AStatistics+milestone%3A%22Milestone+3%22)
Декомпозиция эпика в md: [DecompositionStatistics.md](https://github.com/user-attachments/files/24368895/DecompositionStatistics.md)
Эпик: [Statistics - table teamwork](https://github.com/pavelseleznev/iOS-FakeNFT-StarterProject-Extended-Public/issues/98)

Намеченное и фактическое время представлено в файле, а так же можно посмотреть тут https://github.com/users/pavelseleznev/projects/1/views/3?sliceBy%5Bvalue%5D=urweakness&filterQuery=-label%3ADecompositionFile+assignee%3Aurweakness+milestone%3A%22Milestone+3%22

В NFTDetailView есть слабое место (кнопка перехода на сайт прыгает через раз как .transition(.move(edge: .bottom))). Очень долго мучился в попытках понять причину, прошу помощи.

Сильно торопился все успеть - есть некоторые моменты, которые стоит вынести в ассеты и константы

## Что сделано
- причесал код и перфоманс эпика (+ улучшил синхронизацию загрузок с добавлением кастомных кейсов ошибок с их логированием) (+ добавил кастомные байндинги на пропертис координатора для обработки (добавлено логирование) любых изменений извне)
- добавил файл констант (+ вынес все, что переиспользуется туда)
- локализовал флоу
- реализовал поиск с токенами фильтрации в NFTCollectionView с дебаунсом для поиска
- реализовал фильтры и сортировку в NFTCollectionView (локализовано)
- реализовал search bar на флоу корзины и статистики
- получив добро на изменение дизайна экрана статистики, убрал нумерацию, изменил паддинги, чуть сильнее скруглил углы (сделал флоу более богатым)

- реализовал флоу авторизации на keychain
   1. Чуть изменил дешевый дизайн
   2. Восстановление пароля происходит через отправку foreground push с паролем + пароль копируется в clipboard для дальнейшей вставки
   3. Добавил небольшую искусственную задержку между вызовами подтверждения действия на любом флоу для имитации 
   
- добавил хаптики
- вернул navigation swipe back action при кастомной кнопке

- реализовал NFTDetail
   1. изменил макетную логику показа фулскрин картинок на оверскролл со stretch эффектом (работает как для показа, так и для дизмиза) + добавил не примечательную подсказку свайпа на картинку
   2. во вьюхе списка NFT автора подгружаются все кроме текущей
   3. добавил Divider между информацией об NFT и ценой с кнопкой экшена корзины
   4. добавил тень на Currencies для лучшей видимости в светлой теме
   5. обновление NFT происходит локально в скопе и прокидывается notification через NotificationCenter на изменение кеша и отправки изменений на сервер (AsyncNFTs занимается этим, NFTColllectionView принимает сообщения и пинает AsyncNFTs через колбек)
   6. добавил возможность обновить картинку по ошибке
   7. у стрелок сделал цвета инверсивными по фону, так как в некоторых случаях они сливаются

- реализовал Onboarding
   1. добавил автоскролл с прогрессом
   2. добавил примитивные анимации

- реализовал Splash
   1. добавил оптимизированные бесконечные ленты из картинок на фон (вынужден был использовать UIKit обертку для установки оффсета на скролл контент)
   2. добавил лоадер с набором текстов с маской лоадер шиммера 

- реализовал рейтинг алерт
   1. изменил макетный бекграунд на стекло приятным с градиентом сзади
   2. добавил пустую функцию отправки рейтинга (пока что нет метрики)

## Скриншоты (если UI)

Splash (dark, en)
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/6164d924-e926-49d0-9367-9aa0bee52755" />

Onboarding (dark, en)
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/968ffd3e-88d8-4f11-938d-3e1420d7e460" />

Cart
(light, ru)
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/e2f41c9c-dc2b-4ac6-bd63-57518c06267d" />
(dark, en)
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/ffa98bdb-fcb2-4bf2-a0ee-eb8643c81bf4" />
(dark, en)
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/5f192cab-78bf-426d-a1ff-47203aa81274" />

UsersCollection (dark, en)
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/3b940db3-bb09-4585-9404-f323e32090db" />

NFTCollection (dark, ru)
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/fd2b9fd7-1aae-44a4-bb62-0b1c928e4d66" />
<img width="367" height="357" alt="image" src="https://github.com/user-attachments/assets/96360538-55aa-40d1-a1c0-905d159b99aa" />
<img width="370" height="361" alt="image" src="https://github.com/user-attachments/assets/9685a52b-95c4-4caa-8901-907798bfde3e" />
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/688c288c-50a7-4974-bc3b-abe76e96d353" />
<img width="243" height="98" alt="image" src="https://github.com/user-attachments/assets/715396c3-d571-407d-839a-bcab1e3b1cd6" />
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/44c12df1-546b-4878-85cd-a17d2e3f9189" />

UserProfile (dark, en)
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/205c45ac-1ce4-49be-8782-1aa4acc75f47" />

NFTDetail (dark, en)
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/9f5ea977-0393-479f-86b6-24ce0786e96f" />
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/6bfff4f9-3f02-4c7d-99f6-6e627aaa454d" />
<img width="136.875" height="300" alt="image" src="https://github.com/user-attachments/assets/f0a95584-bffd-4220-ab26-3de8a46a8a7f" />


## Чеклист
- [+] - Код собирается и запускается в Xcode
- [+] - Нет ворнингов
- [+] - Поддержка превью в разных стейтах (если UI)

## Ссылка на задачу/issue
- [Statistics M3](https://github.com/pavelseleznev/iOS-FakeNFT-StarterProject-Extended-Public/issues/98)

urweakness committed:
[Feature] Localization added

urweakness committed:
[Refactor] NFTStorage
- unnecessary async removal

urweakness committed:
[Refactor] DataLoader
- cancellation checking and throwing modified

urweakness committed:
[Feature] NetworkRequest
- POST added

urweakness committed:
[Feature] ObservedNetworkClient
- setCurrency, pay added
- putOrderAndPay -> putOrder renamed

urweakness committed:
[Feature] CurrenciesRequest
- SetCurrnecyByIDRequest added

urweakness committed:
[Feature] OrderRequests
- PayPayload added
- PutOrderAndPayRequest -> PuOrderRequest rename

urweakness committed:
[Feature] CurrencyResponse
- mock added
- Equatable now
- CurrencyContainer added

urweakness committed:
[Feature] CurrencySetResponse
- implemented

urweakness committed:
[Feature] AboutAuthorView
- isTabView for preview added

urweakness committed:
[Feature] NFTService
- data getters added
- cart methods added + modified
- comments rename

urweakness committed:
[Refactor] ObservedNetworkClientTests
- rename

urweakness committed:
[Refactor] system
- scheme localization changed

urweakness committed:
[Refactor] FullScreenCover + Page
- splash screen moved to Page
- Page -> cart pages added

urweakness committed:
[Refactor] Localization added to SortActions

urweakness committed:
[Feature] CartSortActionsViewModifier + Dialogs
- implemented and added to dialogs

urweakness committed:
[Feature] Alerts localized

urweakness committed:
[Feature] Tab
- localized
- cart added

urweakness committed:
[Feature] TabBarView
- appear animation added

urweakness committed:
[Refactor] LoadingShimmerPlaceholderView
- view cornerRadius changed

urweakness committed:
[Feature] RatingPreview
- content hides on skeleton

urweakness committed:
[Feature] ProfileContainer
- localization added

urweakness committed:
[Feature] ProfileHeader
- AsyncImage now supports animation + phase handling
- imageSize now constant

urweakness committed:
[Feature] ProfileActionCell
- localized

urweakness committed:
[Feature] NFTNameRateAuthorView
- skeleton support added

urweakness committed:
[Feature] NFTImageView
- AsyncImage support phase handling + animation
- image switching on error added (infinite try-to-load cycle)
- favourite shadow changed

urweakness committed:
[Feature] NFTCostView
- skeleton support added

urweakness committed:
[Feature] NFTCartImageView
- implemented

urweakness committed:
[Feature] NFTVerticalCell
- skeleton content hiding added

urweakness committed:
[Feature] NFTCartCellView
- skeleton support added
- image view changed
- model changed to NFTModelContainer

urweakness committed:
[Feature] NFTCollectionView
- localized
- scrollTransition animation modified

urweakness committed:
[Feature] AsyncNFTs
- tasks priority added

urweakness committed:
[Feature] StatisticsViewModel
- error handling added with alert

urweakness committed:
[Feature] StatisticsView
- default sort option changed
- every user cell has id now to perform better animations
- error alert added

urweakness committed:
[Feature] StatisticsProfileView
- localized

urweakness committed:
[Feature] StatisticsNFTCollectionView
- localized

urweakness committed:
[Feature] UserListCell
- AsyncImage now support phase handling with animation

urweakness committed:
[Feature] SplashView
- has error alert now
- loads user data
- calls onComplete callback on finish loading to perform main flow

urweakness committed:
[Feature] Coordinator
- path is Page typed now
- loadUserData removal
- onSplashComplete callback added
- popToRoot changed (TabView at 1 index now)
- fullScreenCover splash removal -> splash page added

urweakness committed:
[Feature] CoordinatorView
- splash is root now
- unnecessary load, hitTesting, error alert, checkAuthState removal
- tabView navigation back button restricted

urweakness committed:
[Feature] Cart flow
- implemented

urweakness committed:
[Feature] EmptyContentView
- universal instead of EmptyCartView

urweakness committed:
[Feature] DataLoader
- cancelErrors array added to check localized description to skip cancellation as error

urweakness committed:
[Refactor] CurrencyResponse
- api reality mock refactor

urweakness committed:
[Feature] NFTResponse
- mock imageURLStrings added

urweakness committed:
[Feature] Localizable updated with new strings

urweakness committed:
[Refactor] ObservedNetworkClientTests updated with getUsers payload

urweakness committed:
[Feature] NFTService
- tasks priority added
- clearCart errors handler added

urweakness committed:
[Refactor] Sheet + Page
- nftDetail moved from Sheet to Page
- Page statNFTCollection updated
- Page nftDetail updated

urweakness committed:
[Feature] Stretchy
- implemented

urweakness committed:
[Feature] Skeleton
- implemented

urweakness committed:
[Refactor] SplashView
- tasks priority changed

urweakness committed:
[Feature] PaymentMethodsChooseView
- task priority added

urweakness committed:
[Feature] CartViewModel
- tasks priority added
- unnecessary MainActor.run removed

urweakness committed:
[Refactor] Skeleton applied

urweakness committed:
[Feature] NFTVerticalCell
- skeleton applied
- NFTDetail applied on onTapGesture with didTapDetail callback

urweakness committed:
[Refactor] AsyncNFTs
- loading refactored to background polling
- error presentation on error suuport added
- tasks priority added

urweakness committed:
[Refactor] NFTCollectionView
- emptyNFTsView added
- background polling task lifecycle
- on error changed
- NFTDetail support added

urweakness committed:
[Feature] StatisticsView + ViewModel
- loadingState added
- loading view added
- tasks priority added

urweakness committed:
[Feature] StatisticsNFTCollectionView
- NFTDetail support added

urweakness committed:
[Refactor] StatisticsProfileViewModel
- didTapProfileActionCell NFTDetail support added

urweakness committed:
[Feature] CartView
- tasks priority added
- EmptyCartView -> EmptyContentView changed

urweakness committed:
[Feature] Coordinator
- NFTDetail support added
- NFTDetail moved from Sheet to Page

urweakness committed:
[Feature] NFTDetailView
- reimplemented

urweakness committed:
[Feature] NFTDetail ViewModels
- implemented

urweakness committed:
[Feature] NFTDetail UIComponents
- implemented

urweakness committed:
[Refactor] Assets updated

urweakness committed:
[Feature] Localizable updated for Rating

urweakness committed:
[Feature] Rating implemented

urweakness committed:
[Feature]
- onboarding assets added
- launch screen moved

urweakness committed:
[Refactor] NavigationBackButton
- moved to file

urweakness committed:
[Feature] Localization updated

urweakness committed:
[Feature] Fonts
- bold32 added for OnboardingView

urweakness committed:
[Refactor] Stretchy
- unnecessary condition removal

urweakness committed:
[Feature] TabIndicatorsView
- universal implemented

urweakness committed:
[Refactor] Views
- transition combined
- animation moved to Constants
- lineLimit moved to Constants
- some renaming

urweakness committed:
[Feature] Coordinator + CoordinatorView
- onboarding support added
- app rating support added
- custom bindings added to handle binding changes (Coordinator properties are private now) [using self on getters 'cause of CoordinatorView is app root view]
- back button extension moved to file
 - pop -> backAction rename

urweakness committed:
[Feature] Page
- onboarding added
- hasNotBackButton comp property added

urweakness committed:
[Feature] Constants added

urweakness committed:
[Feature] NFTDetail
- fully implemented

urweakness committed:
[Feaeture] Onboarding
- implemented

urweakness committed:
[Feature] Splash
- featured with infinite carousel nfts

urweakness committed:
[Refactor] NFTDetailCurrneciesView
- shadow added to perform better corners visibility

urweakness committed:
[Refactor] NFTDetailImagesView
- error text wrapped by ZStack to apply dismiss scale

urweakness committed:
[Feature] OnboardingView + TabIndicatorView
- autoscrolling progress added

urweakness committed:
[Feature] Localizable update

urweakness committed:
[Refactor] Strings correct convertation implemented

urweakness committed:
[Feature] LoadingView
- error state with refresh implemented

urweakness committed:
[Refactor] NFTDetailViewModel
- animation from Constants

urweakness committed:
[Feature] NFTDetailGoToSellerSiteButtonView + SwipeSuggestionChevronView
- implemented

urweakness committed:
[Fix] NFTDetailCostWithAddToCartView + NFTDetailToolbarView
- view blinking fix on modelUpdateTriggerID update
- some optimizations

urweakness committed:
[Refactor] NFTDetailScrollViewHandlerView
- optimizations

urweakness committed:
[Feature] NFTDetailImagesView
- drag to dismiss velocity support added
- correct AsyncImagePhase handling
- nftsImagesURLsStrings refactored to [String : String] to allow refresh view by id
- gestures are enabled on full screen image
- gestures priority changed

urweakness committed:
[Fix] NFTDetailView
- unnecessary ScrolLViewReader removed
- on image full screen content transition changed
- unnecessary animations removed
- subviews moved to files
- image swipe suggestion added

urweakness committed:
[Refactor] NFTs update refactor + better perfomance
- AsyncNFTs sends updates on server and performs local storage update now using notifcations to ask AsyncNFTs to update (others views performs updates locally and sends notification)

urweakness committed:
[Feature] SplashLoadingComment + Localization
- new added and olds modified

urweakness committed:
Decomposition file update

numpadCrusader reviewed (approved):
Looks good to me. The code is well-structured and easy to follow. Great work, Timur!

urweakness committed:
KingFisher removed

urweakness committed:
[Feature] Search added with tokens to NFTCollectionView + Localization update

numpadCrusader commented on iOS-FakeNFT-Extended/Modules/UIComponents/NFT/AsyncNFTs.swift:
```diff
@@ -30,6 +32,28 @@ final class AsyncNFTs: ObservableObject {
 			}
 			.map(\.value)
+			.filter { model in
+				guard let model else { return true }
```
можно сделать compactMap, тогда не придется проверять model на nil

urweakness replied:
к сожалению так нельзя, так как compactMap разрушит скелетоны

urweakness committed:
[Refactor] unused NFTModel removal

numpadCrusader reviewed (approved):
Отличная работа! Оставил 1 коммент по возможному улучшению

urweakness committed:
[Feature] AsyncNFTs
- filtration & sorting update + sortComparator + filterApplier
- updateIDs handler guard added to controll updates (stream loads every unloaded ids, so if IDs didnt updated -> there is no nft to update :))
- tokenAction triggering objectWillChange.send() explicitilty beacuse tokens are located on Menu that didnt update view if data changed
- lifecycle clearAllTokens private now + viewDIdDissapeared clears all tasks at all

urweakness committed:
[Feature] NFTCollectionView
- sorting tokens added (localized)
- filter&sorting button added on isSearching (toolbar hides on search is active)
- massive view modifiers incapsulated in private view extension
- value animations moved

urweakness committed:
[Feature] SearchBar
- optinal search bar added to BaseConfirmationDIalogVIewModifier
- shopping cart flow search bar added
- CartView massive modifiers incapsulated in private view extension
- CartViewModel methods privacy refactor and movement to extensions + viewDidDissapear added + unnecessary ArrayDifference removal instead of Set().subtracting to calc difference simplier + cartCostLabel safe string convert
- SellerNFTsViewModel guard added to skip no changed cycle

urweakness committed:
[Refactor] NFTCollectionView + PaymentMethodsChooseView
- payment view shown empty content view on empty or bad internet delay
- nft collection has better scrollTransition appearance now

urweakness committed:
[Refactor] SplashViewModel + StatisticsView
- splash loading first comment randomized
- methods incapsulated

Alex-rev-dev commented on iOS-FakeNFT-Extended/Foundation/NetworkClient/DataLoader.swift:
```diff
@@ -29,13 +30,17 @@ final class DataLoader {
 			return try await operation()
 		} catch {
-			let isCancellation = error.localizedDescription.lowercased().contains("cancelled")
+			let isCancellation =
```
обычно тело переменной не переносят на новую строчку, будет корректнее сделать перенос после ||

Alex-rev-dev commented on iOS-FakeNFT-Extended/Modules/NFTDetail/UIComponents/NFTDetailImagesView.swift:
```diff
@@ -0,0 +1,292 @@
+		velocityHistory.append(rawVelocity)
+		if velocityHistory.count == velocityHistoryCapacity { velocityHistory.removeFirst() }
+		
+		let velocity = velocityHistory.reduce(0, +) / CGFloat(velocityHistory.count)
```
предлагаю добавить проверку деления на 0

Alex-rev-dev commented on iOS-FakeNFT-Extended/Modules/NFTDetail/UIComponents/NFTDetailImagesView.swift:
```diff
@@ -0,0 +1,292 @@
+	var magnificationGesture: some Gesture {
+		MagnificationGesture()
+			.onChanged { value in
+				print(value)
```
этот print точно нужен?

urweakness replied:
нет, уберу, спасибо
дебажил много

Alex-rev-dev commented on iOS-FakeNFT-Extended/Modules/NFTDetail/UIComponents/SellerNFTsView.swift:
```diff
@@ -0,0 +1,146 @@
+	
+	@Previewable @State var excludingNFTID = ""
+	let authorID = "ab33768d-02ac-4f45-9890-7acf503bde54"
+//	let authorID = "ef96b1c3-c495-4de5-b20f-1c1e73122b7d"
```
давай удалим лишний код

urweakness replied:
я изменю название констант

Alex-rev-dev commented on iOS-FakeNFT-Extended/Modules/NFTDetail/ViewModel/SellerNFTsViewModel.swift:
```diff
@@ -0,0 +1,254 @@
+		)
+	}
+	
+	func sendUpdateNotification(
```
давай функции внутри класса сделаем приватными

urweakness replied:
они и так приватные

urweakness commented:
к сожалению, не могу реализовать авторизацию, так как нет апи, а делать моковое апи - не благодарно
разве, что авторизоваться из кейчейна
@Alex-rev-dev

urweakness commented:
если что сделаю авторизацию к мерджу в дев на кейчейне
@Alex-rev-dev

urweakness commented:
как бы я не терзал иконку корзины, она не поддается
 @Alex-rev-dev

urweakness committed:
[Refactor] Views
- last suggestions correction

urweakness commented:
@Alex-rev-dev 
будет ли работа принята если я уберу дешевые элементы и добавлю search bar сверху?
<img width="371" height="776" alt="image" src="https://github.com/user-attachments/assets/5d0d37d7-23da-42ca-aca9-c5e0bb1e6ad2" />

urweakness committed:
[Feature] StatisticsView
- search bar added with search performance
- some search refactor

urweakness committed:
[Feature] AsyncImageCached implementation with usage
- async image loader
- image cache with NSCache<NSString, UIImage> + LRU removing oldest + handling memory warning
- AsyncImageCached instead of AsyncImage
- usage

urweakness committed:
[Feature] NetworkMonitor implementation with usage
- implemented
- preload in App
- DataLoader usage

urweakness committed:
[Feature] Navigation interactive pop bringed back with custom back button

urweakness committed:
[Refactor] PaymentMethodsChooseView
- currency mistake correction

urweakness committed:
[Feature] CartViewModel + StatisticsView/Model
- lang sort priority fix
- Statistics refreshable feature

urweakness committed:
[Feature] NFTCollectionView
- scrollTransition simplified
- layout bottom padding on search for filter button updated

urweakness committed:
[Feature] NFTCartCellView
- real data loading added

urweakness committed:
[Feature] Shimmer skeleton fix

urweakness committed:
[Feature] NFTButtonStyle
- appearance feature

urweakness committed:
[Feature] StatisticsView + UserListCell
- redesign

urweakness committed:
[Feature] Constants + Localizable updated

urweakness committed:
[Feature] Authorization flow
- implemented on keychain (AuthKeychain)
- restore password uses UNUserNotificationCenter foreground pushes (AppDelegate created for foreground handle) and clipboard
- Coordinator - rootPage condition changed + updatePath changed + perform methods added to handle auth flow
- Constants  - defaultAnimation shorten + auth keys added
- Page stable id instead of unstable + auth
- auth localized

urweakness committed:
[Feature] Tab views hides toolbar for better navigation toolbar appearance

urweakness committed:
[Feature] NFTButtonStyle
- isEnabled supported now

urweakness committed:
[Feature] PaymentMethodChooseView
- empty content replacement animation added

urweakness committed:
[Feature + Refactor] Single Responsability AppContainer microservices implementation + usage refactor

urweakness committed:
[Feature] DebouncingViewModel
- onDebounce subscription dismissable now

urweakness committed:
[Feature] AsyncImageCached optimizations

urweakness committed:
[Feature] ObservedNetworkClient
- Sendable methods conformance

urweakness committed:
[Feature] NFTUpdatePayload
- screenID added to skip current updates

urweakness committed:
[Feaeture] Views optimizations + task priority fix

urweakness committed:
[Feature] CartFlow
- background polling separated + optimizations

urweakness committed:
[Feature] AuthFlow
- fields did't clear on dissapear fix + optimizations

urweakness committed:
[Feature] App
- background app data updates implemented with switching between in app long polling with short interval and BGTask by scenePhase

urweakness committed:
[Feature] Services
- reimplmemented with +- single responsability (DRY KISS YAGNI)
- StorageActor with shared storage
- implemented with save to storage & from storage restoration on app reboot
- methods for updateIfNeeded added to use in app refresh cycles

urweakness committed:
[Feature] Navigation
- flow pages splitted by enums + usage
- AppContainer refactor + AppContainerBuilder
- appContainer now builds in App
- screens present logic changed
- new services usage

urweakness committed:
[Refactor] DataLoader cleaned

urweakness committed:
[Fix] LoginTextField unnecessary print removal

urweakness committed:
[Fix] NFTButtonStyle label clickable area fix

urweakness committed:
[Feature] Splash
- unnecessary loading + error removal
- await presentation added
- timer memory leak fix

urweakness committed:
[Refactor] Cart
- onSubmit callback instead
- updates only from storage + update notification from storage
- long polling unloaded nfts in

urweakness committed:
[Refactor] PaymentMethodsChoose
- onComplete callback instead
- updates only from storage
- currencies UI updates acceleration

urweakness committed:
[Refactor] UserListCell
- UX rating added because of rating sorting existence
- small redesign

urweakness committed:
[FIx] Statistics
- statistics rating real-time sorting updates fix
- cell ui updates fix
- rating sorting descending now
- author name line limited for 1

urweakness committed:
[Feature] ProfileStorage updates
- uses shared storage with sync now

urweakness committed:
[Feature] UserListItemResponse mock rating randomed

urweakness committed:
[Feature] NFTDetailCurrenciesView
- loads from storage cache instantly now

urweakness committed:
[Feature] StatisticsProfileView
- optimized by constant

urweakness committed:
[Test] ProfileView simple UI for profileService updates tests

urweakness committed:
[Refactor] NFTDetailScrollViewHandlerView
- calculations moved to parent view
- optimized by PreferenceKey

urweakness committed:
[Feature] NFTDetails opmimizations

urweakness committed:
[Feature] SellersNFTs Stack opmiziation
- stable elements

urweakness committed:
[Feature] NFTDetail SellersNFTs optimization
- now inherits root author loaded nfts (in AsyncNFTs) and passes throw all navigation heirarchy from this author (stable nfts)
- no live updates (needs reentrancy to author NFTCollectionView to reload)

urweakness committed:
[Refactor] Currencies UI/UX
- unnecessary currencies skeletons removal
- currency cells capsuled

urweakness committed:
[Refactor] Rating UI/UX
- RatingPreview invisilble yellow/lightGrey color on light theme changed, shadow removed
- UserListCell capsuled, image/rating frame minimized
- ProfileHeaeder rating scaled up for better visibility

urweakness committed:
[Refactor] BaseConfirmationDialogViewModifier searchBar UI/UX
- moved to bottom of custom safeArea top toolbar
- searchBar shadow removed
- searchBar bg color changed like system

urweakness committed:
[Refactor] Page path debugDescription added instead of full print

urweakness committed:
[Refactor] Notification.Name mistakes correction

urweakness committed:
[Feature] Splash infinite carousel recreated wth TimelineView with offset caclulation
- max optimizations with drawingGroup + shared timeline context

urweakness committed:
[Feature] SplashView
- BlinkingAppAcon added
- splash duration increased
- dissapear animations added
- unnecessary appContainer dependency removed from Splash

urweakness committed:
[Fix] SafeAreTopBackground blur low opacity fix

urweakness committed:
[Fix] NFTCollectionView
- optional init for NFTDetail flow due to bad API that didn't show authorID from Catalog flow (Collections/NFTs has their own ids, but its not a authorID) -> goodbye :[[[[[

urweakness committed:
[Feature] Haptics implementation + usage

urweakness committed:
[Refactor] BlinkingAppIcon moved to shared UIComponents

urweakness committed:
[Feature] NFTImageVIew simplified + haptic on like
